src/main/java/task_tree_UI/TaskTreeUI.java

```diff
@@ -21,25 +22,37 @@
  *
  * @author Aayush Bhan
  */
-
-
-public class TaskTreeUI extends CentralWindow {
+public class TaskTreeUI extends CentralWindow implements CurriculumDependentWindow {
     private JTree taskTree;
     private int displayId;
+    private final String idDisplayText = " (ID n.o.: ";
     private JPanel mainPanel;
+    private final JButton addTaskButton;
+    private final JButton deleteTaskButton;
+    private final DisplayTaskTreeController displayController;
+    private final WindowManager programWindows;
 
 
     /**
      * Default constructor for TaskTree UI
      */
-    public TaskTreeUI() {
+    public TaskTreeUI(WindowManager windowManager, DisplayTaskTreeController displayController) {
+        this.programWindows = windowManager;
```

Comment:
You also have to add this window to the windowManager, and create a reference key for it, so other windows can access it